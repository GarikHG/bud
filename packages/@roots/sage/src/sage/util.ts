import {Sage} from './interface'

export function dependencyConditional(
  this: Sage,
  dependencies: string[],
): boolean {
  const project = this.discovery.projectInfo

  return (
    dependencies.filter(
      (dep: string): boolean =>
        Object.keys(project.dependencies)?.includes(dep) ||
        Object.keys(project.devDependencies)?.includes(dep),
    ).length > 0
  )
}
