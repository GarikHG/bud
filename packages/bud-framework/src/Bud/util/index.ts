export * from './pretty'
export * from './format'
export * from './logger'
export * from './proxy'
export * from './callMeMaybe'
