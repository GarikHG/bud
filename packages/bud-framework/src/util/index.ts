import fab from './fab'
import {notify} from './notify'

import dd from './dd'
import dump from './dump'
import format from './format'
import highlight from './highlight'
import pretty from './pretty'
import {fs} from './fs'
import {logger} from './logger'
import {os} from './os'
import terminate from './terminate'
import processHandler from './processHandler'

import __ from 'lodash'
import {__ as Lo} from 'lodash'

const lo: Lo = __

export {
  lo,
  pretty,
  fs,
  dump,
  dd,
  logger,
  fab,
  format,
  highlight,
  processHandler,
  notify,
  terminate,
  os,
}
