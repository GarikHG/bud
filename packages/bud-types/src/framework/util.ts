import {Loose} from '../base'

export interface Dump {
  function(obj: any, prettierOptions?: any): void
}

export interface Fab {
  false: () => boolean
  true: () => boolean
  undefined: () => undefined
  null: () => null
}

export type Format = {
  [key: string]: Formatter
}
export type Formatter = (any) => string

interface TerminateOptions extends Loose {
  dump?: boolean
  timeout?: number
}

export type Terminate = (
  options?: TerminateOptions,
) => () => (err: Error) => void

export interface Util extends Loose {
  Terminate
  Formatter
  Fab
}
