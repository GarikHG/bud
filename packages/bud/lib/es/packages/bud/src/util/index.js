/**
 * @roots/bud v.2.0.0-rc.7 {@link https://roots.io/bud}
 *
 * A friendly build tool to help manage your project assets.
 *
 * Issues? {@link https://github.com/roots/bud/issues}
 *
 * Consider funding our work üôèüèΩ {@link https://github.com/sponsors/roots}
 *
 * @copyright 2020 Roots {@link https://roots.io}
 * @license MIT
 */
import { shortCircuit } from './shortCircuit.js';
import { dump } from './dump.js';
import { fab } from './fab.js';
import { projectRoot } from './projectRoot.js';
import { terminate } from './terminate.js';
import { processHandler } from './processHandler.js';
import { fs } from './fs.js';
import { usedExt } from './usedExt.js';
export { logger } from './logger.js';

var util = {
    fs: fs,
    dump: dump,
    shortCircuit: shortCircuit,
    fab: fab,
    projectRoot: projectRoot,
    processHandler: processHandler,
    terminate: terminate,
    usedExt: usedExt,
};

export { util };
