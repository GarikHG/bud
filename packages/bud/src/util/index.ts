import {dump} from './dump'
import {fab} from './fab'
import {projectRoot} from './projectRoot'
import {shortCircuit} from './shortCircuit'
import {terminate} from './terminate'
import {processHandler} from './processHandler'
import {fs} from './fs'
import {usedExt} from './usedExt'
import type {Util} from './types'

export const util: Util = {
  fs,
  dump,
  shortCircuit,
  fab,
  projectRoot,
  processHandler,
  terminate,
  usedExt,
}

export {logger} from './logger'
