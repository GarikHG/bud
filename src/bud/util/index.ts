import {dump} from './dump'
import {except} from './except'
import {fab} from './fab'
import {shortCircuit} from './shortCircuit'

export const util: Util = {
  dump,
  except,
  shortCircuit,
  fab,
}

export type Dump = (obj: Object) => void
export type Except = Function;
export type ShortCircuit = () => any
export type Fab = {
  false: () => boolean
  true: () => boolean
  undefined: () => undefined
  null: () => null
}
export type Util = {
  dump: Dump,
  except: Except,
  shortCircuit: ShortCircuit,
  fab: Fab,
}
