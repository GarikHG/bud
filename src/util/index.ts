import {dump} from './dump'
import {except} from './except'
import {fab} from './fab'
import {projectRoot} from './projectRoot'
import {shortCircuit} from './shortCircuit'
import {terminate} from './terminate'
import {processHandler} from './processHandler'
import {usedExt} from './usedExt'
import {fs} from './fs'
import type {Util} from './types'

export const util: Util = {
  fs,
  dump,
  except,
  shortCircuit,
  fab,
  projectRoot,
  processHandler,
  terminate,
  usedExt,
}

export {nodeExternals} from './nodeExternals'
export {logger} from './logger'
