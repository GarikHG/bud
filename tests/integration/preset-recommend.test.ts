import {Project} from '../util/integration'

jest.setTimeout(60000)

describe('examples/preset-recommend', () => {
  let project: Project

  beforeAll(async () => {
    project = new Project({
      name: 'preset-recommend',
      dir: 'examples/preset-recommend',
    })

    await project.setup()
  })

  it('[main.js] has contents', () => {
    expect(project.assets['main.js'].length).toBeGreaterThan(10)
  })

  it('[main.js] is transpiled', () => {
    expect(
      project.assets['main.js'].includes('import'),
    ).toBeFalsy()
  })

  it('[main.css] is transpiled', () => {
    expect(project.assets['main.css']).toMatchSnapshot()
  })
})
