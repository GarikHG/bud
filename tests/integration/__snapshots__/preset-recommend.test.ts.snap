// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/preset-recommend [app.css] is transpiled 1`] = `"h2 .class{margin-top:5rem}"`;
